libuavcan/test/transport/can_acceptance_filter_configurator.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>,
+ *                    Ilia  Sheremet <illia.sheremet@gmail.com>
+ */
+
+#include <gtest/gtest.h>
+#include <cassert>
+
+#include <uavcan/transport/can_acceptance_filter_configurator.hpp>
+#include "../node/test_node.hpp"
+#include "uavcan/node/subscriber.hpp"
+#include <uavcan/equipment/camera_gimbal/AngularCommand.hpp>
+#include <uavcan/equipment/air_data/Sideslip.hpp>
+#include <uavcan/equipment/air_data/TrueAirspeed.hpp>
+#include <uavcan/equipment/air_data/AngleOfAttack.hpp>
+#include <uavcan/equipment/ahrs/AHRS.hpp>
+#include <uavcan/equipment/air_data/StaticPressure.hpp>
+#include <uavcan/protocol/file/BeginFirmwareUpdate.hpp>
+#include <uavcan/node/service_client.hpp>
+#include <uavcan/node/service_server.hpp>
+#include <iostream>
+#include <bitset>
+
+#if UAVCAN_CPP_VERSION >= UAVCAN_CPP11
+
+template <typename DataType>
+struct SubscriptionListener
+{
+    typedef uavcan::ReceivedDataStructure<DataType> ReceivedDataStructure;
+
+    struct ReceivedDataStructureCopy
+    {
+        uavcan::MonotonicTime ts_monotonic;
+        uavcan::UtcTime ts_utc;
+        uavcan::TransferType transfer_type;
+        uavcan::TransferID transfer_id;
+        uavcan::NodeID src_node_id;
+        uavcan::uint8_t iface_index;
+        DataType msg;
+
+        ReceivedDataStructureCopy(const ReceivedDataStructure& s)
+        : ts_monotonic(s.getMonotonicTimestamp())
+        , ts_utc(s.getUtcTimestamp())
+        , transfer_type(s.getTransferType())
+        , transfer_id(s.getTransferID())
+        , src_node_id(s.getSrcNodeID())
+        , iface_index(s.getIfaceIndex())
+        , msg(s)
```

Comment:
almost =) that's how uncrustify do:
        ReceivedDataStructureCopy(const ReceivedDataStructure& s) :
            ts_monotonic(s.getMonotonicTimestamp()), ts_utc(s.getUtcTimestamp()), transfer_type(s.getTransferType()),
            transfer_id(s.getTransferID()), src_node_id(s.getSrcNodeID()), iface_index(s.getIfaceIndex()), msg(s)
        { }
just putting everything in 2 lines. My bad, I corrected it a bit wrong